osu.Game/Screens/SelectV2/SoloSongSelect.cs

```diff
@@ -20,10 +28,49 @@ public partial class SoloSongSelect : SongSelect
         private PlayerLoader? playerLoader;
         private IReadOnlyList<Mod>? modsAtGameplayStart;
 
+        [Resolved]
+        private BeatmapSetOverlay? beatmapOverlay { get; set; }
+
+        [Resolved]
+        private BeatmapManager beatmaps { get; set; } = null!;
+
+        [Resolved]
+        private IAPIProvider api { get; set; } = null!;
+
         [Resolved]
         private INotificationOverlay? notifications { get; set; }
 
-        public override bool EditingAllowed => true;
+        [Resolved]
+        private IDialogOverlay? dialogOverlay { get; set; }
+
+        [Resolved]
+        private OsuGame? game { get; set; }
+
+        public override IEnumerable<OsuMenuItem> GetForwardActions(BeatmapInfo beatmap)
+        {
+            yield return new OsuMenuItem(ButtonSystemStrings.Play.ToSentence(), MenuItemType.Highlighted, () => SelectAndStart(beatmap)) { Icon = FontAwesome.Solid.Check };
+            yield return new OsuMenuItem(ButtonSystemStrings.Edit.ToSentence(), MenuItemType.Standard, () => edit(beatmap)) { Icon = FontAwesome.Solid.PencilAlt };
+
+            yield return new OsuMenuItemSpacer();
+
+            if (beatmap.OnlineID > 0)
+            {
+                yield return new OsuMenuItem("Details...", MenuItemType.Standard, () => beatmapOverlay?.FetchAndShowBeatmap(beatmap.OnlineID));
+
+                if (beatmap.GetOnlineURL(api, Ruleset.Value) is string url)
+                    yield return new OsuMenuItem(CommonStrings.CopyLink, MenuItemType.Standard, () => game?.CopyToClipboard(url));
+
+                yield return new OsuMenuItemSpacer();
+            }
+
+            // TODO: replace with "remove from played" button when beatmap is already played.
+            yield return new OsuMenuItem(SongSelectStrings.MarkAsPlayed, MenuItemType.Standard, () => beatmaps.MarkPlayed(beatmap)) { Icon = FontAwesome.Solid.TimesCircle };
+            yield return new OsuMenuItem(SongSelectStrings.ClearAllLocalScores, MenuItemType.Standard, () => dialogOverlay?.Push(new BeatmapClearScoresDialog(beatmap)))
+            {
+                Icon = FontAwesome.Solid.Eraser
+            };
+            yield return new OsuMenuItem(WebCommonStrings.ButtonsHide.ToSentence(), MenuItemType.Destructive, () => beatmaps.Hide(beatmap));
+        }
```

Comment:
Compared to [v1](https://github.com/ppy/osu/blob/faa73f5d04fdc6be1fe3d8921bbc793523e7eb0b/osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmap.cs#L287-L310), appears to be missing collections management